internal/logprocessor/logprocessor.go

```diff
@@ -60,8 +61,12 @@ func (c *Client) parseResponse(body []byte) ([]byte, error) {
 	}
 
 	conn := connection{}
-
-	conn.SrcIP = rb.XFwdFor
+	if strings.Contains(rb.XFwdFor, ",") {
+		parsedIP := strings.Split(rb.XFwdFor, ",")
+		conn.SrcIP = parsedIP[0]
+	} else {
+		conn.SrcIP = rb.XFwdFor
+	}
 	validIP := net.ParseIP(rb.XFwdFor)
```

Comment:
Shouldn't this check happen as part of the above check? Making sure its a valid ip after splitting on a comma? It looks like you set conn.SrcIP to the parsed ip, then try to validate the original again (which may contain the ','s again).

Reply:
Thats kinda what I was thinking as well, I just couldnt put it into words...